### PR TITLE
Fix links

### PR DESCRIPTION
**Pull request by severo**
Found with

```
muffet --rate-limit 5 -i "https://huggingface.co/docs/hub/*" https://huggingface.co/docs/hub/
```

### CODE REVIEW

**Review by julien-c — COMMENTED** [review]
let's wait for @mishig25 review though because i think those links used to work

**Comment by severo** [comment]
The detail: 
- in https://huggingface.co/docs/hub/datasets-dask, the link "Login with your Hugging Face account" works: https://huggingface.co/docs/huggingface_hub/quick-start#login
- but in https://huggingface.co/docs/hub/en/datasets-dask, it becomes https://huggingface.co/docs/hub/huggingface_hub/quick-start which is a 404

**Review by mishig25 — APPROVED** [review]
@severo thanks a lot for finding this problem.

Explanation why this problem is occurring.
1. We've used relative links such as `[🤗 Datasets](../datasets/index)` to switch from hub-docs page to another library's page.
2. This relative path works fine when there is no `{lang}` or `{branch}` path present in the URL.

Example:
`[🤗 Datasets](../datasets/index)` was going to different paths depending whether `{lang}` was present or not:
* /docs/hub/datasets-dask -> /docs/datasets/index **(good)**
* /docs/hub/en/datasets-dask -> /docs/hub/datasets/index **(bad)**

To clarify, using relative links is not a problem when you are linking pages within a same library (since they will share same `{lang}` or `{branch}` prefixes). Using relative links is a problem when linking pages from different libraries. Therefore, we should always use absolute path `/docs/{lib}/{page}` when linking pages from different libraries.

File: docs/hub/datasets-adding.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ The rich features set in the `huggingface_hub` library allows you to manage repo
 
 ## Using other libraries
 
-Some libraries like [🤗 Datasets](../datasets/index), [Pandas](https://pandas.pydata.org/), [Polars](https://pola.rs), [Dask](https://www.dask.org/) or [DuckDB](https://duckdb.org/) can upload files to the Hub.
+Some libraries like [🤗 Datasets](https://huggingface.co/docs/datasets/index), [Pandas](https://pandas.pydata.org/), [Polars](https://pola.rs), [Dask](https://www.dask.org/) or [DuckDB](https://duckdb.org/) can upload files to the Hub.
```
Comment (mishig25) [inline docs/hub/datasets-adding.md:70]:
```suggestion
Some libraries like [🤗 Datasets](/docs/datasets/index), [Pandas](https://pandas.pydata.org/), [Polars](https://pola.rs), [Dask](https://www.dask.org/) or [DuckDB](https://duckdb.org/) can upload files to the Hub.
```

small nit: we can just do `/docs/**`

Comment (severo) [reply docs/hub/datasets-adding.md:70]: OK, better! I'll change

Comment (mishig25) [reply docs/hub/datasets-adding.md]: yep should apply to other diffs as well :)

**Comment by severo** [comment]
Done. I changed `https://huggingface.co/docs` for `/docs` (but only in the datasets doc pages... I let the rest of the docs untouched)

**Comment by pcuenca** [comment]
Thanks for the clarification @mishig25!

**Pull request merged**